static/js/components/inputs/DateField.js

```diff
@@ -24,6 +25,7 @@ export default class DateField extends React.Component {
     keySet: Array<string>,
     label: string,
     omitDay: boolean,
+    futureYear: boolean,
```

Comment:
Can you rename to `allowFutureYear` here and in `boundDateField` to make it more explicit?